marshmallow_dataclass/__init__.py

```diff
@@ -405,13 +405,46 @@ def field_for_schema(
             metadata["required"] = False
             return field_for_schema(subtyp, metadata=metadata, base_schema=base_schema)
         elif typing_inspect.is_union_type(typ):
-            subfields = [
-                field_for_schema(subtyp, metadata=metadata, base_schema=base_schema)
-                for subtyp in arguments
-            ]
-            import marshmallow_union
 
-            return marshmallow_union.Union(subfields, **metadata)
+            def deserialization_disambiguation(obj_dict, base_dict):
```

Comment:
can we move these functions and SchemaPolyfieldProxy to a separate file (polyfield.py? union.py?) exposing a single method `PolyField/Union(**metadata)` which will hide away the complexity of the union case?